### PR TITLE
fix(conan): correct regex for packages without @user/channel but with revisions

### PR DESCRIPTION
**Pull request by dazze1812**
## Changes

Fixes the problem with Conan packages that have no @user/channel but a revision set.

## Context

see https://github.com/renovatebot/renovate/discussions/26673

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/conan/__fixtures__/conanfile.txt [hunk 1]
```diff
@@ -1,16 +1,17 @@
 [requires]
 poco/1.9.4
-zlib/[~1.2.3, loose=False]  
+zlib/[~1.2.3, loose=False]
```
Comment (viceice) [inline lib/modules/manager/conan/__fixtures__/conanfile.txt:3]: what's the change here and why?

File: lib/modules/manager/conan/__fixtures__/conanfile.txt [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 fake/8.62.134@test/dev
+cairo/1.17.2#aff2d03608351db075ec1348a3afc9ff
 cairo/1.17.2@_/_#aff2d03608351db075ec1348a3afc9ff
 
 [build_requires]
 7zip/[>1.1 <2.1, include_prerelease=True]
 curl/[~1.2.3, loose=False, include_prerelease=True]@test/dev
 boost/[>1.1 <2.1]
-catch2/[2.8]     
+catch2/[2.8]
```
Comment (viceice) [inline lib/modules/manager/conan/__fixtures__/conanfile.txt:12]: same

File: lib/modules/manager/conan/__fixtures__/conanfile.txt [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 openssl/[~=3.0]@test/prod
-cmake/[>1.1 || 0.8] 
+cmake/[>1.1 || 0.8]
```
Comment (viceice) [inline lib/modules/manager/conan/__fixtures__/conanfile.txt:14]: same

File: lib/modules/manager/conan/__fixtures__/conanfile.txt [hunk 4]
```diff
@@ -29,4 +30,4 @@ openssl:shared=True
 
 [imports]
 bin, *.dll -> ./bin # Copies all dll files from packages bin folder to my local "bin" folder
-lib, *.dylib* -> ./bin # Copies all dylib files from packages lib folder to my local "bin" folder
\ No newline at end of file
+lib, *.dylib* -> ./bin # Copies all dylib files from packages lib folder to my local "bin" folder
```
Comment (viceice) [inline lib/modules/manager/conan/__fixtures__/conanfile.txt:33]: same

**Review by viceice — COMMENTED** [review]
you need to mark PR as ready to trigger full CI run

File: lib/modules/manager/conan/__fixtures__/conanfile.txt (hunk 1 above)
Comment (dazze1812) [reply lib/modules/manager/conan/__fixtures__/conanfile.txt:3]: Sorry. This was the code formatter. The trailing spaces were removed. Shall I revert this?

File: lib/modules/manager/conan/__fixtures__/conanfile.txt (hunk 2 above)
Comment (dazze1812) [reply lib/modules/manager/conan/__fixtures__/conanfile.txt:12]: Sorry. This was the code formatter. The trailing spaces were removed. Shall I revert this?

File: lib/modules/manager/conan/__fixtures__/conanfile.txt (hunk 3 above)
Comment (dazze1812) [reply lib/modules/manager/conan/__fixtures__/conanfile.txt:14]: Sorry. This was the code formatter. The trailing spaces were removed. Shall I revert this?

File: lib/modules/manager/conan/__fixtures__/conanfile.txt (hunk 4 above)
Comment (dazze1812) [reply lib/modules/manager/conan/__fixtures__/conanfile.txt:33]: Sorry. This was the code formatter. Shall I revert this?

File: lib/modules/manager/conan/__fixtures__/conanfile.txt (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/conan/__fixtures__/conanfile.txt:3]: no, that's fine. didn't show up on mobile app

**Review by viceice — APPROVED** [review]

**Pull request merged**